daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc/IdeState.hs

```diff
@@ -0,0 +1,56 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Cli.Damlc.IdeState
+    ( IdeState
+    , getIdeState
+    , withIdeState
+    ) where
+
+import qualified Language.Haskell.LSP.Messages as LSP
+
+import DA.Daml.GHC.Compiler.Options
+import qualified DA.Service.Logger as Logger
+import qualified DA.Service.Daml.Compiler.Impl.Scenario as Scenario
+import Development.IDE.Core.Rules.Daml
+import qualified Development.IDE.Core.Shake as Shake
+import Development.IDE.State.API
+import qualified Development.IDE.Types.Logger as IdeLogger
+
+getIdeState
```

Comment:
Half wonder if this should be `getDamlIdeState`.

Reply:
Good idea! Changed